### PR TITLE
Typed guardian shutdown race

### LINKED ISSUE #24172: Adapted typed actor system guardian stop races with adding shutdown hooks
If you create an adapted actor system (will soon be the only way to create a typed actor sytem) and your guardian stops right away it'll race with the coordinated shutdown addings its hooks e.g.

```                                                                                                                                            
[info]   java.util.concurrent.RejectedExecutionException: ActorSystem already terminated.                                                                                                                             
[info]   at akka.actor.ActorSystemImpl$TerminationCallbacks.addRec$1(ActorSystem.scala:989)                                                                                                                                                                                                                                                                                                                                          
[info]   at akka.actor.ActorSystemImpl$TerminationCallbacks.add(ActorSystem.scala:993)                                                                                                                                
[info]   at akka.actor.ActorSystemImpl.registerOnTermination(ActorSystem.scala:812)                                                                                                                                   
[info]   at akka.actor.ActorSystemImpl.registerOnTermination(ActorSystem.scala:811)                                                                                                                                   
[info]   at akka.actor.CoordinatedShutdown$.createExtension(CoordinatedShutdown.scala:163)                                                                                                                                                                                                                                 
[info]   at akka.actor.CoordinatedShutdown$.createExtension(CoordinatedShutdown.scala:34)                                                                                                                             
[info]   at akka.actor.ActorSystemImpl.registerExtension(ActorSystem.scala:881)                                                                                                                                       
[info]   at akka.actor.ExtensionId.apply(Extension.scala:78)                                                                                                                                                          
[info]   at akka.actor.ExtensionId.apply$(Extension.scala:77)                                                                                                                                                         
[info]   at akka.actor.CoordinatedShutdown$.apply(CoordinatedShutdown.scala:34)  
```

### PR DESCRIPTION
**Pull request by johanandren**
Fixes #24172

### CODE REVIEW

File: akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/adapter/AdapterSpec.scala [hunk 1]
```diff
@@ -157,6 +157,33 @@ class AdapterSpec extends AkkaSpec {
...
+      }
+    }
+
+    "not crash if guardian is stopped very quickly" in {
+      for { _ ← 0 to 10 } {
+        var system: akka.actor.typed.ActorSystem[Done] = null
+        try {
+          system = ActorSystem.create(Actor.immutable[Done] { (ctx, done) ⇒
+            Actor.stopped
+          }, "AdapterSpec-stopping-guardian-2")
```
Comment (johanandren) [inline akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/adapter/AdapterSpec.scala]: Ach, maybe someone should send it a `Done` also 😆

**Comment by johanandren** [comment]
I guess I should add a test in Akka-actor for this as well, with a custom guardian that stops right away as that is where the actual problem was.

File: akka-actor/src/main/scala/akka/actor/ActorRefProvider.scala [hunk 2]
```diff
@@ -646,10 +654,14 @@ private[akka] class LocalActorRefProvider private[akka] (
   private[akka] def init(_system: ActorSystemImpl) {
     system = _system
     rootGuardian.start()
+    eventStream.startDefaultLoggers(_system)
+  }
+
+  private[akka] def initDone(): Unit = {
     // chain death watchers so that killing guardian stops the application
+    // but don't do that until the system is completely started or else things will explode
```
Comment (chbatey) [inline akka-actor/src/main/scala/akka/actor/ActorRefProvider.scala]: :boom:

**Comment by johanandren** [comment]
One does not simply change the actor system init logic.

**Comment by johanandren** [comment]
Can't get `RemoteInitErrorSpec` to pass will continue with that next week.

**Comment by johanandren** [comment]
PLS BUILD

**Comment by johanandren** [comment]
Maybe we shouldn't fix this but instead take a step back (see #24279 )

**Comment by johanandren** [comment]
Closing this as the other solution is much cleaner.

**Pull request closed without merging**